src/server.cpp

```diff
@@ -3583,3 +3596,51 @@ void dedicated_server_loop(Server &server, bool &kill)
 			server.m_bind_addr.getPort());
 #endif
 }
+
+/*
+ * Mod channels
+ */
+
+
+bool Server::joinModChannel(const std::string &channel)
+{
+	return m_modchannel_mgr->leave_channel(channel, PEER_ID_SERVER);
+}
+
+bool Server::leaveModChannel(const std::string &channel)
+{
+	return m_modchannel_mgr->join_channel(channel, PEER_ID_SERVER);
+}
+
+bool Server::sendModChannelMessage(const std::string &channel, const std::string &message)
+{
+	if (!m_modchannel_mgr->channel_registered(channel))
+		return false;
+
+	broadcastModChannelMessage(channel, message, PEER_ID_SERVER);
+	return true;
+}
+
+void Server::broadcastModChannelMessage(const std::string &channel,
+		const std::string &message, u16 from_peer)
+{
+	const auto &peers = m_modchannel_mgr->get_channel_peers(channel);
+	if (peers.empty()) {
+		return;
+	}
+
+	NetworkPacket resp_pkt(TOCLIENT_MODCHANNEL_MSG,
+			2 + channel.size() + 2 + message.size());
+	resp_pkt << channel << message;
+	for (u16 peer_id : peers) {
+		// Ignore sender
+		if (peer_id == from_peer)
+			continue;
```

Comment:
How about inter-CSM communication on the same client? I think we should broadcast it to every client if it's accepted content.

Reply:
for others, see discussion here: http://irc.minetest.net/minetest-dev/2017-09-01#i_5062210